components/sup/src/manager/service/hooks.rs

```diff
@@ -66,30 +66,60 @@ pub trait Hook: fmt::Debug + Sized {
 
     fn file_name() -> &'static str;
 
+    fn deprecated_file_name() -> Option<&'static str> {
+        None
+    }
+
+    /// Tries to load a hook if a (deprecated) hook file exists.
+    ///
+    /// Returns the hook if template file (deprecated or not) is found
     fn load<C, T>(service_group: &ServiceGroup, concrete_path: C, template_path: T) -> Option<Self>
     where
         C: AsRef<Path>,
         T: AsRef<Path>,
     {
-        let concrete = concrete_path.as_ref().join(Self::file_name());
-        let template = template_path.as_ref().join(Self::file_name());
-        match std::fs::metadata(&template) {
-            Ok(_) => {
-                let pair = match RenderPair::new(concrete, &template) {
-                    Ok(pair) => pair,
-                    Err(err) => {
-                        outputln!(preamble service_group, "Failed to load hook: {}", err);
-                        return None;
-                    }
-                };
-                Some(Self::new(service_group, pair))
-            }
-            Err(_) => {
-                debug!(
-                    "{} not found at {}, not loading",
-                    Self::file_name(),
-                    template.display()
+        let file_name = Self::file_name();
+        let deprecated_file_name = Self::deprecated_file_name();
+
+        let concrete = concrete_path.as_ref().join(&file_name);
+        let template = template_path.as_ref().join(&file_name);
+        let deprecated_template = deprecated_file_name
+            .as_ref()
+            .map(|n| template_path.as_ref().join(n));
+
+        let has_template = template.exists();
+        let has_deprecated_template = deprecated_template.as_ref().map_or(false, |t| t.exists());
+
+        let template_to_use = if has_template {
+            if has_deprecated_template {
+                outputln!(preamble service_group, 
```

Comment:
Looks like there's a `rustfmt` issue here that [travis caught](https://travis-ci.org/habitat-sh/habitat/jobs/455237768):
```
internal error: left behind trailing whitespace
  --> stdin:95
   |
95 |                 outputln!(preamble service_group, 
   |                                                  ^
```